src/lib/Microsoft.Health.Fhir.Ingest.Template/CalculatedFunction/CalculatedFunctionContentTemplateFactory.cs

```diff
@@ -0,0 +1,95 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using DevLab.JmesPath;
+using EnsureThat;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.Ingest.Template.CalculatedFunction
+{
+    public class CalculatedFunctionContentTemplateFactory : HandlerProxyTemplateFactory<TemplateContainer, IContentTemplate>
+    {
+        private const string TargetTypeName = "CalculatedContent";
+        private JmesPath _jmesPath;
+
+        public CalculatedFunctionContentTemplateFactory()
+        {
+            /*
+             * TODO Load and register additional custom JmesPath functions. For now, simply create the basic JmesPath object
+             */
+            _jmesPath = new JmesPath();
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Globalization", "CA1303:Do not pass literals as localized parameters", Justification = "Exception message")]
+        public override IContentTemplate Create(TemplateContainer jsonTemplate)
+        {
+            EnsureArg.IsNotNull(jsonTemplate, nameof(jsonTemplate));
+
+            if (!jsonTemplate.MatchTemplateName(TargetTypeName))
+            {
+                return null;
+            }
+
+            if (jsonTemplate.Template?.Type != JTokenType.Object)
+            {
+                throw new InvalidTemplateException($"Expected an object for the template property value for template type {TargetTypeName}.");
+            }
+
+            var calculatedFunctionTemplate = jsonTemplate.Template.ToValidTemplate<CalculatedFunctionContentTemplate>();
+            var measurementExtractor = new MeasurementExtractor(calculatedFunctionTemplate, CreateExpressionEvaluatorFactory(calculatedFunctionTemplate));
+            return measurementExtractor;
+        }
+
+        private IExpressionEvaluatorFactory CreateExpressionEvaluatorFactory(CalculatedFunctionContentTemplate template)
+        {
+            var evaluatorCache = new Dictionary<string, IExpressionEvaluator>();
+            IExpressionEvaluatorFactory expressionEvaluator = new ExpressionEvaluatorFactory(_jmesPath, template.DefaultExpressionLanguage);
```

Comment:
This might be better as a dependency injected to aid with testing.

Reply:
Sure thing. After working in the code base I think that we could benefit from a general refactor to move more of our initialization code into the DI framework. For example, `CollectionContentTemplateFactory` could be a DI Singleton versus referencing it via the static class instance as we do today. In fact it would be nice to do what @namalu did for the Workspace Platform repo and refactor each application into its own HostedService. I can do that in a follow on PR, as this one is getting a bit big.

Reply:
Done